target/linux/realtek/files-5.10/drivers/clk/realtek/clk-rtl83xx.c

```diff
@@ -484,6 +481,26 @@ static const struct clk_ops rtcl_clk_ops = {
 	.recalc_rate = rtcl_recalc_rate,
 };
 
+static void rtcl_unlock_registers(void)
+{
+	u32 enable, reg;
+
+	if (rtcl_ccu->soc != RTCL_SOC838X)
+		return;
+/*
+ * Some RTL838X devices are shipped with register access locked. In this case
+ * we cannot read and/or write to LXB or SW PLL registers. As there is no real
+ * benefit of always unlocking/locking these registers just open up everything.
+ */
+	enable = RTL838X_INT_RW_CTRL_READ_EN | RTL838X_INT_RW_CTRL_WRITE_EN;
+	reg = ioread32((void *)RTL838X_INT_RW_CTRL);
+	if ((reg & enable) != enable) {
+		reg |= enable;
+		iowrite32(reg, (void *)RTL838X_INT_RW_CTRL);
+		pr_warn("%s: registers unlocked\n", __func__);
```

Comment:
Why should I warn about unlocked registers? That is the desired state and if set correctly by someone else (vendor) everything is fine. Opening up the registers should have a meaningful level.

Reply:
That's exactly my point. "unlocked registers" is the desired state, so that's what you set. "locked registers" is just the default state, so not necessarily an indication that something is wrong, which would warrant a `pr_warn()`. I don't think you need to inform the user at all, just unlock the registers and be done with it.